security/fma-supervisor.md

```diff
@@ -0,0 +1,215 @@
+- Recovery
+    - With respect to the Node, an arbitrary amount of re-sync may be required.
+    - Recovery to the network depends entirely on the impact of the node outage.
+
+## FM4a: Managed Nodes are at Different Heights
+
+- Description
+    - The Supervisor is managing one Managed Node per chain, and using the data reported from their derivation to calculate cross-safety.
+    - While the Node for Chain A has derived to some height (L1 block 100), the Node for Chain B is still processing earlier L1 blocks (L1 block 90).
+    - During this period, cross safety can’t be fully calculated between beyond L1 block 90 for *any chain which references Chain B* when the referenced data is yet unsynced (direct or indirect).
+    - Any chains who do not need Chain B data of the unsynced region can process normally.
+- Risk Assessment
+    - Zero Impact, Certainty.
+    - This is just the natural consequence of syncing from multiple data sources at once, you can’t know what you don’t yet know.
+    - The Supervisor already takes this lack of data into account when responding to queries and advancing safety.
+
+## FM4b: A Managed Node Stalls or Lags Significantly
+
+- Description
+    - The Supervisor is managing one Managed Node per chain, and using the data reported from their derivation to calculate cross-safety.
+    - For some reason, a Managed Node is not able to sync the chain quickly, or at all (perhaps the node is down or is failing)
+    - Without the data from this Node, the Supervisor cannot advance safety for any other chain which depends on this stalled cahin.
+    - The Supervisor also won’t be able to answer any validity questions about the un-synced portion of the chain (which is why safety doesn’t advance)
+    - Assuming they take some dependency on un-syncing chain, other chains will eventually stall their cross-unsafe and cross-safe chains
+- Risk Assessment
+    - Low Impact, Low likelihood
+    - The Supervisor knows the boundaries of data it can/can’t report on, and won’t answer *incorrectly* just because it doesn’t have data.
+    - If the Managed Nodes are Sequencers who need interop data to build blocks, they will be unable to validate interop messages, and will therefore not include them in block building
+- Mitigation
+    - Supervisor supports a feature in which *multiple* Nodes of a given chain can be connected as Managed Nodes. If one Node goes down, syncing can continue with the backup.
+        - This feature is mostly ready in the happy path, but there are known gaps in managing secondary Managed Nodes during block replacements and resets.
+        - This feature *also* needs much more robust testing in lifelike environments. Previous development cycles were spent in the devnet trying to enable this feature, which was slow and risky. To get this feature working well, we need to leverage our new network-wide testing abilities.
+
+## FM5: Increased Operational/Infrastructure Load Results in Fewer Nodes
+
```

Comment:
I think we should add some RPC controls to the sequencer to rate-limit or temporarily entirely cancel inclusion of cross-chain messages, so we can use it if there's ever an issue with load or security.

Reply:
I did add this in the AIs at the bottom